packages/node/src/methods/state-channel/deposit/operation.ts

```diff
@@ -173,22 +189,31 @@ async function getDepositContext(
   params: Node.DepositParams,
   publicIdentifier: string,
   provider: BaseProvider,
-  networkContext: NetworkContext
+  networkContext: NetworkContext,
+  tokenAddress: string
 ): Promise<DepositContext> {
   const { multisigAddress } = params;
-  const threshold = await provider.getBalance(multisigAddress);
+  const threshold =
+    tokenAddress === AddressZero
```

Comment:
You have to do this kind of if else logic a lot in this PR. Try to abstract these situations it into a class of some kind